src/components/AnalysisPage/ThicknessAnalysisBox/ThickAnalysisBox.jsx

```diff
@@ -1,21 +1,29 @@
-import { useState } from "react";
+import { useState, useEffect } from "react";
 import { Row, Col, InputNumber, Typography, Segmented } from "antd";
 const { Text } = Typography;
 import "./ThickAnalysisBox.css";
 import DisplayStatistic from "../DisplayStatistic";
 import StartButton from "./StartButton/StartButton";
 import PropTypes from "prop-types";
 
-const ThickAnalysisBox = ({ partId, thickData }) => {
+const ThickAnalysisBox = ({
+  partId,
+  thickData,
+  partsThresholdHistory,
+  setPartsThresholdHistory,
+}) => {
   const [thresholdValue, setThresholdValue] = useState(null);
   const [thinSurfaceArea, setThinSurfaceArea] = useState(null);
   const [isThin, setIsThin] = useState(null);
 
   console.log(thickData);
 
-  const partsThresholdHistory = thickData
-    ? thickData.map((data) => data.threshold)
-    : [];
+  useEffect(() => {
+    if (thickData) {
+      setPartsThresholdHistory(thickData.map((data) => data.threshold));
```

Comment:
Can the thresholdHistory stay as a piece of state in ThicknessAnalysisBox rather than having to be stored up in AnalysisPage? I think that would be cleaner if so.

Reply:
And actually do we even need threhsoldHIstory? Or can we get the same info by just using thickness data state?

Reply:
Yeah we can do that!!!

I think after my latest commit on the thickness continuation branch, it opened up all the possibilities .

Because we immediately have the latest threshold options available.

I just quickly tried it and it worked